crates/rover-client/src/blocking/client.rs

```diff
@@ -39,33 +42,66 @@ impl GraphQLClient {
         header_map: &HashMap<String, String>,
     ) -> Result<Q::ResponseData, RoverClientError> {
         let header_map = build_headers(header_map)?;
-        let response = self.execute::<Q>(variables, header_map)?;
+        let request_body = self.get_request_body::<Q>(variables)?;
+        let response = self.execute(&request_body, header_map)?;
         GraphQLClient::handle_response::<Q>(response)
     }
 
-    pub(crate) fn execute<Q: GraphQLQuery>(
+    pub(crate) fn get_request_body<Q: GraphQLQuery>(
         &self,
         variables: Q::Variables,
+    ) -> Result<String, RoverClientError> {
+        let body = Q::build_query(variables);
+        Ok(serde_json::to_string(&body)?)
+    }
+
+    pub(crate) fn execute(
+        &self,
+        request_body: &str,
         header_map: HeaderMap,
     ) -> Result<Response, RoverClientError> {
-        let body = Q::build_query(variables);
         tracing::trace!(request_headers = ?header_map);
-        tracing::debug!("Request Body: {}", serde_json::to_string(&body)?);
-        self.client
-            .post(&self.graphql_endpoint)
-            .headers(header_map)
-            .json(&body)
-            .send()
-            .map_err(|e| {
-                if e.is_connect() {
+        tracing::debug!("Request Body: {}", request_body);
+        let graphql_operation = || {
+            let response = self
+                .client
+                .post(&self.graphql_endpoint)
+                .headers(header_map.clone())
```

Comment:
I had to add this `.clone()` due to `FnMut` requirements

Reply:
I think you can put `move` in front of the closure instead.

Reply:
ah yeah maybe no you can't because it's an Fn not an FnOnce :sweat_smile: never mind then

Reply:
yeah the move doesn't seem to help us out here :/